CMakeLists.txt

```diff
@@ -91,6 +91,8 @@ set(ZIG_STATIC off CACHE BOOL "Attempt to build a static zig executable (not com
 set(ZIG_SHARED_LLVM off CACHE BOOL "Prefer linking against shared LLVM libraries")
 set(ZIG_STATIC_LLVM off CACHE BOOL "Prefer linking against static LLVM libraries")
 set(ZIG_STATIC_ZLIB off CACHE BOOL "Prefer linking against static zlib")
+set(ZIG_ENABLE_ZSTD on CACHE BOOL "Enable linking zstd")
+set(ZIG_ENABLE_LIBCPP on CACHE BOOL "Enable linking libcpp")
```

Comment:
I'd like to avoid adding these if possible. It looks like `ZIG_ENABLE_LIBCPP` is unused and can simply be deleted.

As for `ZIG_ENABLE_ZSTD`, it will confuse package maintainers and users into believing that Zig depends on zstd, which is not the case. In reality, it is a transitive dependency that we only have to care about when LLVM is built statically, and is built with zstd enabled. In such case, the user must pass `-DZIG_STATIC_ZSTD=ON`. In all other cases, the user can leave the defaults as-is.

I don't see the purpose of an additional option.

Reply:
The purpose of `ZIG_ENABLE_ZSTD` was to get around `ZIG_STATIC` enabling `ZIG_STATIC_ZSTD`, which would cause a find_library search, which would then fail in cases where it didn't exist (like building zig for the host during bootstrapping). 

I changed the way that the defaults were set to allow `ZIG_STATIC_ZSTD` to be set explicitly off when `ZIG_STATIC` is enabled. The `build.bat` script will set this to `OFF` explicitly when building for the host.